pkg/sql/create_table.go

```diff
@@ -128,8 +128,10 @@ func getTableCreateParams(
 		params.ExecCfg().Settings, dbID, tableName)
 	if isTemporary {
 		if !params.SessionData().TempTablesEnabled {
-			return nil, 0, unimplemented.NewWithIssuef(5807,
-				"temporary tables are unsupported")
+			return nil, 0, unimplemented.NewWithIssuef(
+				46260,
+				"temporary tables require experimental_enable_temp_tables to be set to 'on'",
```

Comment:
I would keep the original message, then pass your new sentence as a hint (`errors.WithHint`).

Reply:
unfortunately the issue number is already the HINT. how do you see the error appearing (ignoring the infrastructure)?

Reply:
"unfortunately" - how so?

You can have multiple consecutive hints.

Reply:
hurray! fortune favours me!